### PR TITLE
Require mutate.update to be a cocina object.

### LINKED ISSUE #263: Require mutate.update params to be a cocina model
Previously it allowed params to be a hash; however, with the current code hashes will no longer work. An error should be raised if not a cocina model.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #263

## Why was this change made? 🤔
Surface errors when invoking this method.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: lib/dor/services/client/objects.rb [hunk 1]
```diff
@@ -8,9 +8,9 @@ class Client
       # API calls that are about a repository objects
       class Objects < VersionedService
         # Creates a new object in DOR
-        # @param params [Cocina::Models::RequestDRO,Cocina::Models::RequestCollection,Cocina::Models::RequestAPO]
+        # @param params [Cocina::Models::RequestDRO,Cocina::Models::RequestCollection,Cocina::Models::RequestAdminPolicy]
```
Comment (justinlittman) [inline lib/dor/services/client/objects.rb:11]: Fixed some incorrect method docs.

**Review by jmartin-sul — APPROVED** [review]

File: lib/dor/services/client/mutate.rb [hunk 2]
```diff
@@ -33,7 +33,10 @@ def apply_admin_policy_defaults
         # @raise [BadRequestError] when ETag not provided.
         # @return [Cocina::Models::DROWithMetadata,Cocina::Models::CollectionWithMetadata,Cocina::Models::AdminPolicyWithMetadata] the returned model
         # rubocop:disable Metrics/AbcSize
+        # rubocop:disable Metrics/MethodLength
         def update(params:, skip_lock: false)
+          raise ArgumentError, 'Cocina object not provided.' unless params.respond_to?(:externalIdentifier)
```
Comment (jmartin-sul) [inline lib/dor/services/client/mutate.rb:38]: hmm, i just merged, but after glancing back at this again, i'm wondering if `cocina_object` or similar might be a more informative parameter name that `params`

Comment (justinlittman) [reply lib/dor/services/client/mutate.rb:38]: It entirely would, but I don't want to change the method signature at this time.

Comment (jmartin-sul) [reply lib/dor/services/client/mutate.rb:38]: totally fair, and tbh i wasn't thinking about the caller's use of named args when i made that observation.

**Pull request merged**